plugins/automation/app/services/discourse_automation/destroy.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class DiscourseAutomation::Destroy
+  include Service::Base
+
+  # @!method self.call(guardian:, params:)
+  #   @param [Guardian] guardian
+  #   @param [Hash] params
+  #   @option params [Integer] :automation_id
+  #   @return [Service::Base::Context]
+
+  policy :can_destroy_automation
+  params do
```

Comment:
All the steps are grouped together, I know you don’t like it for the ones having blocks, it’s not to annoy you, I swear :sweat_smile: I saw that originally you added blank lines around `params`, but not around `transaction`. One thing I’d like for us to do is defining a consistent rule on this. This way, we could create a rubocop rule that will apply this and possibly even autocorrect the format.

So it boils down to choosing between having something like:
```ruby
  policy :can_destroy_automation
  params do
    attribute :automation_id, :integer
    validates :automation_id, presence: true
  end
  model :automation
  transaction do
    step :log_action
    step :destroy_automation
  end
```
versus
```ruby
  policy :can_destroy_automation

  params do
    attribute :automation_id, :integer
    validates :automation_id, presence: true
  end

  model :automation

  transaction do
    step :log_action
    step :destroy_automation
  end
```
I prefer the first version, I think you prefer the second one, but my goal is to have consistency so, in the end, I won’t care that much.

Reply:
Yes I very much prefer the second one, but feel free to make a quick poll in dev-xp channel, no need to ask everyone

Reply:
Ok so the second one won :sweat_smile: